hyperspy/signal_tools.py

```diff
@@ -244,6 +249,191 @@ def apply(self):
         self.span_selector_switch(on=True)
         self.last_calibration_stored = True
 
+@add_gui_method(toolkey="hyperspy.EELSSpectrum.print_edges_table")
+class EdgesRange(SpanSelectorInSignal1D):
+    units = t.Unicode()
+    edges_list = t.Tuple()
+    only_major = t.Bool()
+    order = t.Unicode('closest')
+    complementary = t.Bool(True)
+
+    def __init__(self, signal):
+        super(EdgesRange, self).__init__(signal)
+        if signal.axes_manager.signal_dimension != 1:
+            raise SignalDimensionError(
+                signal.axes_manager.signal_dimension, 1)
+        self.units = self.axis.units
+        self.active_edges = []
+        self.active_complementary_edges = []
+        self.slp = SpectrumLabelPosition(self.signal)
+        self.btns = []
+
+        self._get_edges_info_within_energy_axis()
+
+        self.signal.axes_manager.events.indices_changed.connect(self._on_nav_change,
+                                                                [])
```

Comment:
When connecting, we also need to connect the "disconnection" when closing. For example:

https://github.com/hyperspy/hyperspy/blob/3ec3018f59cee4c9db023408b09d51024fe53d42/hyperspy/signal_tools.py#L616-L621

Reply:
now it connects the 'disconnection' when `signal_plot` is closed